compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/collectors/components/ErrorNodeDiagnosticCollectorComponent.kt

```diff
@@ -64,18 +67,22 @@ class ErrorNodeDiagnosticCollectorComponent(
 
         // If the receiver cannot be resolved, we skip reporting any further problems for this call.
         if (qualifiedAccessOrAnnotationCall is FirQualifiedAccess) {
-            if (qualifiedAccessOrAnnotationCall.dispatchReceiver.hasUnresolvedNameError() ||
-                qualifiedAccessOrAnnotationCall.extensionReceiver.hasUnresolvedNameError() ||
-                qualifiedAccessOrAnnotationCall.explicitReceiver.hasUnresolvedNameError()
+            if (qualifiedAccessOrAnnotationCall.dispatchReceiver.cannotBeResolved() ||
+                qualifiedAccessOrAnnotationCall.extensionReceiver.cannotBeResolved() ||
+                qualifiedAccessOrAnnotationCall.explicitReceiver.cannotBeResolved()
             ) return
         }
 
         reportFirDiagnostic(errorNamedReference.diagnostic, source, reporter, data, qualifiedAccessOrAnnotationCall?.source)
     }
 
-    private fun FirExpression?.hasUnresolvedNameError(): Boolean {
-        return when ((this?.typeRef as? FirErrorTypeRef)?.diagnostic) {
-            is ConeUnresolvedNameError -> true
+    private fun FirExpression?.cannotBeResolved(): Boolean {
```

Comment:
I think `cannotBeResolved` is a better name in this case because it's indeed what the function is about. On the other hand,`skipTrailingDiagnostics` describes what the caller is doing. It's applicable to more than qualified access expressions. For example `a.unresolvedCall().anotherCall()`, we should not report a redundant unresolved reference on `anotherCall`.